File: compiler/python/state-buffer-estimate/layers/tanhlayer.py
from utils.consts  import  *
from .layer         import Layer
from .activlayer    import ActivLayer
import nets.network

##########################################################
class TanhLayer(ActivLayer):
    #-----------------------------------------------------------------
    def __init__(self, param, prev_layer):
        assert(isinstance(prev_layer, Layer))
        super().__init__(param , prev_layer)

    #-----------------------------------------------------------------
    def gJson(self):
        x = super().gJson()
        return x

    #-----------------------------------------------------------------
    def __str__(self):
        baseLayer = self.gBaseLayerStr()
        return ("Tanh " + baseLayer + self.gStateSizesStr())


    #-----------------------------------------------------------------
    @classmethod
    def gTypeStr(cls):
        return "Tanh"

    #-----------------------------------------------------------------
    def qPassThrough(self):
        return False

    #-----------------------------------------------------------------
    def qTanhLayer(self):
        return True

File: compiler/kcc/layers/src/tanhlayer.cpp
#include <sstream>

#include "datatype.hpp"
#include "network.hpp"
#include "tanhlayer.hpp"



namespace kcc {
namespace layers {

//--------------------------------------------------------
TanhLayer::TanhLayer(const Params& params, Layer* prev_layer)
    : ActivLayer(params, prev_layer)
{}

#if 0
//--------------------------------------------------------
TanhLayer::gJson()
{
    x = super().gJson()
    return x
}

//--------------------------------------------------------
TanhLayer* 
TanhLayer::constructFromJson(cls, layerDict, nn)
{
        layerName = Layer.gLayerNameFromJson(layerDict)
        ofmapDesc = Layer.gOfmapDescFromJson(layerDict, nn)
        batch = 1
        param = Layer.Param(layerName, batch, nn)
        prevLayers = Layer.gPrevLayersFromJson(layerDict, nn)
        assert isinstance(prevLayers, list) and len(prevLayers)==1
        layer = TanhLayer(param, prevLayers[0])
        return layer
}
#endif


//--------------------------------------------------------
string
TanhLayer::gString() const
{
    string baseLayerStr = gBaseLayerStr();
    string retVal;
    retVal = retVal + gTypeStr() + " " + baseLayerStr + gStateSizesStr();
    return retVal;
}


const char*
TanhLayer::gTypeStr() const
{
    return "Tanh";
}

bool
TanhLayer::qPassThrough() const
{
    return false;
}

bool 
TanhLayer::qTanhLayer() const
{
    return true;
}

} // namespace layers
} // namespace kcc


